demos/device_defender_for_aws/defender_demo.c

```diff
@@ -0,0 +1,649 @@
+                                                  &( numOpenTcpPorts ) );
+
+        if( metricsCollectorStatus != MetricsCollectorSuccess )
+        {
+            IotLogError( "GetOpenTcpPorts failed. Status: %d.",
+                        metricsCollectorStatus );
+        }
+    }
+
+    /* Collect a list of open UDP ports. */
+    if( metricsCollectorStatus == MetricsCollectorSuccess )
+    {
+        metricsCollectorStatus = GetOpenUdpPorts( &( openUdpPorts[ 0 ] ),
+                                                  OPEN_UDP_PORTS_ARRAY_SIZE,
+                                                  &( numOpenUdpPorts ) );
+
+        if( metricsCollectorStatus != MetricsCollectorSuccess )
+        {
+            LogError( ( "GetOpenUdpPorts failed. Status: %d.",
+                        metricsCollectorStatus ) );
+        }
+    }
+		
+    /* Collect a list of established connections. */
+    if( metricsCollectorStatus == MetricsCollectorSuccess )
+    {
+        metricsCollectorStatus = GetEstablishedConnections( &( establishedConnections[ 0 ] ),
+                                                            ESTABLISHED_CONNECTIONS_ARRAY_SIZE,
+                                                            &( numEstablishedConnections ) );
+
+        if( metricsCollectorStatus != MetricsCollectorSuccess )
+        {
+            IotLogError( "GetEstablishedConnections failed. Status: %d.",
+                        metricsCollectorStatus );
+        }
+    }
+
+    /* Populate device metrics. */
+    if( metricsCollectorStatus == MetricsCollectorSuccess )
+    {
+        status = EXIT_SUCCESS;
+        deviceMetrics.pNetworkStats = &( networkStats );
+        deviceMetrics.pOpenTcpPortsArray = &( openTcpPorts[ 0 ] );
+        deviceMetrics.openTcpPortsArrayLength = numOpenTcpPorts;
+        deviceMetrics.pOpenUdpPortsArray = &( openUdpPorts[ 0 ] );
+        deviceMetrics.openUdpPortsArrayLength = numOpenUdpPorts;
+        deviceMetrics.pEstablishedConnectionsArray = &( establishedConnections[ 0 ] );
+        deviceMetrics.establishedConnectionsArrayLength = numEstablishedConnections;
+    }
+
+    
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static bool subscribeToDefenderTopics( MQTTContext_t * pxMqttContext )
+{
+    bool status = false;
+
+    status = SubscribeToTopic( pxMqttContext,
+                               DEFENDER_API_JSON_ACCEPTED( THING_NAME ),
+                               DEFENDER_API_LENGTH_JSON_ACCEPTED( THING_NAME_LENGTH ) );
+
+    if( status == true )
+    {
+        status = SubscribeToTopic( pxMqttContext,
+                                   DEFENDER_API_JSON_REJECTED( THING_NAME ),
+                                   DEFENDER_API_LENGTH_JSON_REJECTED( THING_NAME_LENGTH ) );
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static int unsubscribeFromDefenderTopics( MQTTContext_t * pxMqttContext )
```

Comment:
ditto

Reply:
Made a commit to update and align the function commit type with demo helper.
e9d96e2